### DISCUSSION TITLE
httfs support for HTTP[S] proxy and custom SSL cert bundle

### DISCUSSION THREAD

**Discussion by oxlade39** [duckdb/duckdb#5944] [Ideas]
I'm currently trying to use DuckDB in a corporate environment where our AWS access goes via an HTTP[s] proxy.
This means the proxy intercepts the HTTPS traffic and requires that we use a custom SSL certificate bundle.
In the case of tools such boto, it requires us setting the `AWS_CA_BUNDLE`.

There doesn't appear to be a way to configure the httpfs to support a custom CA bundle and I think it also might not honor the `HTTP_PROXY`, `HTTPS_PROXY` and `no_proxy` environment variables.

I will attach a notebook that demonstrates the error I see, which is:
```
---------------------------------------------------------------------------
Error                                     Traceback (most recent call last)
Cell In[7], line 1
----> 1 result = con.execute("select * from 's3://fiaim-all-trp-dev-shared/portfolio/AIALPHAGGB|9888/2018-05-18/results.parquet'")

Error: Invalid Error: HTTP HEAD error on '[https://s3.amazonaws.com/fiaim-all-trp-dev-shared/portfolio/AIALPHAGGB|9888/2018-05-18/results.parquet](https://s3.amazonaws.com/fiaim-all-trp-dev-shared/portfolio/AIALPHAGGB%7C9888/2018-05-18/results.parquet)' (Error code 8)
```

**Comment by oxlade39** [comment]
Uploading notebook as zip (ipynb not supported)
[duckdb.zip](https://github.com/duckdb/duckdb/files/10456040/duckdb.zip)

**Comment by Mytherin** [comment]
Possibly related to #3836?

**Reply by oxlade39** [reply]
> Possibly related to #3836?

Thanks. It's definitely related in the sense of the environment variables. Although my issue relates to the httpfs code itself. In my case, specifically when talking to s3 over https.

**Reply by Mytherin** [reply]
Could you perhaps file an issue then?

**Reply by oxlade39** [reply]
> Could you perhaps file an issue then?

Sure, sorry. I thought it might be a feature request rather than an issue so went that route but happy to move this over to an issue.

**Reply by oxlade39** [reply]
can be closed, moved to #6064
